src-tauri/src/graph_p.rs

```diff
@@ -314,6 +331,26 @@ impl Graph {
         edges
     }
 
+    pub fn get_neighbors(&self, node: String) -> Vec<Node> {
+        let node_weight = self.node_idx_map.get(&node).unwrap();
+        let mut neighbors = Vec::new();
+        for neighbor in self.g.neighbors(node_weight.clone()) {
+            neighbors.push(self.g.node_weight(neighbor).unwrap().clone());
```

Comment:
What does this `node_weight` function do? It might be worth adding some quick docs on each graph function if it's not too much of a hassle ([reference](https://doc.rust-lang.org/stable/rust-by-example/meta/doc.html)).